### PR TITLE
Adds LocalSpanCollector

### PR DESCRIPTION
**Pull request by adriancole**
LocalSpanCollector submits spans to a local storage component.

For example, if you are tracing a storage system like cassandra,
you might prefer to write directly to cassandra as opposed to over http.

This code replaces a similar class in zipkin-java, and also supports
cassandra tracing.

### CODE REVIEW

**Comment by adriancole** [comment]
cc @michaelsembwever

File: brave-core/src/main/java/com/github/kristofa/brave/FlushingSpanCollector.java [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.ScheduledExecutorService;
+
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+/**
+ * Extend this class to offload the task of reporting spans to separate thread. By doing so, callers
+ * are protected from latency or exceptions possible when exporting spans out of process.
+ */
+public abstract class FlushingSpanCollector implements SpanCollector, Flushable, Closeable {
```
Comment (adriancole) [inline brave-core/src/main/java/com/github/kristofa/brave/FlushingSpanCollector.java]: This used to be AbstractSpanCollector

File: brave-core/src/main/java/com/github/kristofa/brave/FlushingSpanCollector.java [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+      }
+    }
+  }
+
+  /**
+   * Reports a list of spans over the current transport.
+   *
+   * @throws IOException (or RuntimeException) when thrown, drop metrics will increment accordingly
+   */
+  protected abstract void reportSpans(List<Span> drained) throws IOException;
```
Comment (adriancole) [inline brave-core/src/main/java/com/github/kristofa/brave/FlushingSpanCollector.java]: Main difference here as opposed to what was AbstractSpanCollector is that we don't require subtypes to encode the drained spans

**Comment by adriancole** [comment]
anybody have feelings about this either way? I'm interested in this as I'd like to help make a cassandra tracer with it (that can self-report), as well I'd like to decommission similar code in zipkin https://github.com/openzipkin/zipkin-java/blob/master/zipkin-server/src/main/java/zipkin/server/brave/LocalSpanCollector.java

File: brave-core/src/main/java/com/github/kristofa/brave/FlushingSpanCollector.java (hunk 1 above)
Comment (eirslett) [inline brave-core/src/main/java/com/github/kristofa/brave/FlushingSpanCollector.java]: Will it break anything in the public API?

Comment (adriancole) [inline brave-core/src/main/java/com/github/kristofa/brave/FlushingSpanCollector.java]: >
> Will it break anything in the public API?
>
Nope. main reason for making this a superclass of AbstractCollector was to
prevent that.

**Comment by eirslett** [comment]
LGTM!

**Pull request merged**